xrayvision/visibility.py

```diff
@@ -5,10 +5,408 @@
+
+    @property
+    @abc.abstractmethod
+    def vis_labels(self) -> Union[Iterable[str], None]:
+        """
+        Labels of each visibility.
+        """
+
+    @property
+    @abc.abstractmethod
+    def instrument(self) -> Union[str, None]:
+        """
+        The name of the instrument or observer that measured the visibilities.
+        """
+
+
+class VisibilitiesABC(abc.ABC):
+    @property
+    @abc.abstractmethod
+    def visibilities(self) -> Iterable[apu.Quantity]:
+        """
+        Complex numbers representing the visibilities.
+        """
+
+    @property
+    @abc.abstractmethod
+    def u(self) -> Iterable[apu.Quantity]:
+        """
+        u-coordinate on the complex plane of the visibilities.
+        """
+
+    @property
+    @abc.abstractmethod
+    def v(self) -> Iterable[apu.Quantity]:
+        """
+        v-coordinate on the complex plane of the visibilities.
+        """
+
+    @property
+    @abc.abstractmethod
+    def phase_center(self) -> apu.Quantity[apu.deg]:
+        """
+        The position of the phase center of the visibilities.
+        """
+
+    @property
+    @abc.abstractmethod
+    def meta(self) -> VisMetaABC:
+        """
+        Metadata.
+        """
+
+    @property
+    @abc.abstractmethod
+    def uncertainty(self) -> Union[Iterable[apu.Quantity], None]:
+        """
+        Uncertainties on visibilities values.
+        """
+
+    @property
+    @abc.abstractmethod
+    def amplitude(self) -> Union[Iterable[apu.Quantity], None]:
+        """
+        Amplitudes of the visibilities.
+        """
+
+    @property
+    @abc.abstractmethod
+    def amplitude_uncertainty(self) -> Union[Iterable[apu.Quantity], None]:
+        """
+        Amplitude uncertainty of the visibilities.
+        """
+
+    @property
+    @abc.abstractmethod
+    def phase(self) -> Union[Iterable[apu.Quantity[apu.deg]], None]:
+        """
+        Phases of the visibilities.
+        """
+
+    @property
+    @abc.abstractmethod
+    def phase_uncertainty(self) -> Union[Iterable[apu.Quantity[apu.deg]], None]:
+        """
+        Phase uncertainty of the visibilities.
+        """
+
+
+class Visibilities(VisibilitiesABC):
+    @apu.quantity_input()
+    def __init__(
+        self,
+        visibilities: apu.Quantity,
+        u: apu.Quantity[1 / apu.deg],
+        v: apu.Quantity[1 / apu.deg],
+        phase_center: Union[apu.Quantity[apu.deg], None] = None,
+        meta: Any = dict(),
+        uncertainty: Union[apu.Quantity, None] = None,
+        amplitude: Union[apu.Quantity, None] = None,
+        amplitude_uncertainty: Union[apu.Quantity, None] = None,
+        phase: Union[apu.Quantity[apu.deg], None] = None,
+        phase_uncertainty: Union[apu.Quantity[apu.deg], None] = None,
+    ):
+        r"""
+        A class for holding visibilities.
+
+        Parameters
+        ----------
+        visibilities : `astropy.units.Quantity`
+            Array of N complex visibilities at coordinates in `uv`.
```

Comment:
Once #58 is merge the type will be taken from the type annotations so these can all be
```
       visibilities :
            Array of N complex visibilities at coordinates in `uv`.
```